homeassistant/components/homekit/__init__.py

```diff
@@ -0,0 +1,114 @@
+"""Support for Apple Homekit.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/homekit/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (
+    ATTR_FRIENDLY_NAME, ATTR_SUPPORTED_FEATURES, CONF_PATH, CONF_PORT,
+    EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.util import get_local_ip
+from homeassistant.util.decorator import Registry
+
+TYPES = Registry()
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'homekit'
+REQUIREMENTS = ['HAP-python==1.1.2']
+
+BRIDGE_NAME = 'Home Assistant'
+CONF_PIN_CODE = 'pincode'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.All({
+        vol.Optional(CONF_PORT, default=51826): vol.Coerce(int),
+        vol.Optional(CONF_PIN_CODE, default='123-45-678'): cv.string,
+        vol.Optional(CONF_PATH, default='accessory.state'): cv.string,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Setup the homekit component."""
+    # pylint: disable=unused-variable
+    from .covers import Window  # noqa F401
+    # pylint: disable=unused-variable
+    from .sensors import TemperatureSensor # noqa F401
+
+    conf = config[DOMAIN]
+    port = conf.get(CONF_PORT)
+    pin = str.encode(conf.get(CONF_PIN_CODE))
+    path = hass.config.path(conf.get(CONF_PATH))
```

Comment:
It should not be configurable and it should also be placed inside the Home Assistant config path.

```python
STATE_PATH = '.homekit.state'
path = hass.config.path(STATE_PATH)

```